### PR TITLE
Media Picker - upload button disabled bug #13383

### PR DESCRIPTION
**Pull request by cheeseytoastie**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes (https://github.com/umbraco/Umbraco-CMS/issues/13383)

### Description
Issue is around how the folder path is "got" on the initialisation - if it doesn't come from localstorage (which is wiped on logout- hence how I could recreate). The function gotoFolder(folder) has entityResource.getAncestors that builds the folder path if it's not set (which it isn't on first init) but this isn't on a promise or awaited.

Modified the code that follows so it can only run once this returns by moving it to a separate function performGotoFolder(folder). It then has the path and on first open the editor can add images.

I've tested the following scenarios and it seems to work (without breaking everything).

Media picker with start node set.
Media picker without start node
User with limited media folders (e.g. can't see the start node). Seems to have resolved the issue.

To recreate (without fix). Create a media picker with a start node. Log out, load the media picker and you'll see the grey media picker where the editor cannot upload an image. But if the close and reopen it works. 

After the fix this issue has gone. [🤞](https://emojipedia.org/crossed-fingers)

### CODE REVIEW

**Comment by cheeseytoastie** [comment]
Could we pretty please have this fix in v10. Would make a lot of my editors happy.

**Comment by lauraneto** [comment]
Hi @cheeseytoastie ,

Thank you for the pull request! 🙌
Someone from the Core Collaborators team will have a look at it soon.

Regarding v10, if the initial issue mentions v10 usually the fix is cherrypicked and included in the following v10 release! 🙂

File: src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/mediapicker/mediapicker.controller.js [hunk 1]
```diff
@@ -264,18 +264,24 @@ angular.module("umbraco")
                                 function (f) {
                                     return f.path.indexOf($scope.startNodeId) !== -1;
                                 });
+							folder.path = $scope.path[0].path;
```
Comment (emmagarland) [inline src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/mediapicker/mediapicker.controller.js:267]: Minor point, but is the tabbing slightly out on this method?

Comment (cheeseytoastie) [reply src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/mediapicker/mediapicker.controller.js:267]: Dropped this one. But hopefully the fix can still be merged in?

**Comment by nul800sebastiaan** [comment]
Thanks for this one @cheeseytoastie! Tests well and happy to relieve an annoyance 🙌

**Pull request merged**